frontend/src/components/utilities/secrets/getSecretsForProject.ts

```diff
@@ -29,8 +29,8 @@ interface SecretProps {
 
 interface FunctionProps {
   env: string;
-  setIsKeyAvailable: any;
-  setData: any;
+  setIsKeyAvailable?: any;
+  setData?: any;
```

Comment:
imo, `setIsKeyAvailable` and `setData` are unnecessary here. we can return `{ secrets, isKeyAvailable }` and set state after getting the return value instead of passing these functions.
what do you think @mv-turtle ?

Reply:
I agree! Just need to make sure we check that it works in all the cases. We should definitely try to optimize this!